pkg/local_object_storage/metabase/exists.go

```diff
@@ -86,14 +93,50 @@ func (db *DB) exists(tx *bbolt.Tx, addr oid.Address) (exists bool, err error) {
 	return firstIrregularObjectType(tx, cnr, objKey) != objectSDK.TypeRegular, nil
 }
 
-// inGraveyard returns:
-//  * 0 if object is not marked for deletion;
+var errStopIteration = errors.New("stop")
+
+// objectStatus returns:
+//  * 0 if object is available;
 //  * 1 if object with GC mark;
-//  * 2 if object is covered with tombstone.
-func inGraveyard(tx *bbolt.Tx, addr oid.Address) uint8 {
+//  * 2 if object is covered with tombstone;
+//  * 3 if object is expired.
+func objectStatus(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) uint8 {
+	// we check only if the object is expired in the current
+	// epoch since it is considered the only corner case: the
+	// GC is expected to collect all the objects that have
+	// expired previously for less than the one epoch duration
+
+	rawOID := []byte(addr.Object().EncodeToString())
+	var expired bool
+
+	// bucket with objects that have expiration attr
+	expirationBucket := tx.Bucket(attributeBucketName(addr.Container(), objectV2.SysAttributeExpEpoch))
+	if expirationBucket != nil {
+		// bucket that contains objects that expire in the current epoch
+		currEpochBkt := expirationBucket.Bucket([]byte(strconv.FormatUint(currEpoch, 10)))
```

Comment:
What about the objects that expired in the previous epoch?

Reply:
i've added comment to that func:

```
// we check only if the object is expired in the current
// epoch since it is considered the only corner case: the
// GC is expected to collect all the objects that have
// expired previously for less than the one epoch duration
```

i expect that GC is able to collect all the expired objects in one epoch (one hour currently), so the only problem objects are the object that expire in the current epoch but have not been collected by the GC yet. theoretically, we could check more than one epoch (we do not have index that could say what objects are expired, only the objects that has *exact* expiration epoch), so we should know how deep to check OR we should add new expiration index

i thought about force new epoch event that could lead to some problems. we may add some threshold for that cases (e.g. check if an object was not expired in the last `5` epoch)

what do you think?